ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt

```diff
@@ -26,27 +26,53 @@ internal sealed class OfferingSelection {
 class PaywallOptions(builder: Builder) {
 
     internal val offeringSelection: OfferingSelection
-    private val shouldDisplayDismissButton: Boolean
+    internal val shouldDisplayDismissButton: Boolean
     val fontProvider: FontProvider?
     val listener: PaywallListener?
-    internal var mode: PaywallMode = PaywallMode.default
+    internal val mode: PaywallMode
     val dismissRequest: () -> Unit
 
     init {
         this.offeringSelection = builder.offeringSelection
         this.shouldDisplayDismissButton = builder.shouldDisplayDismissButton
         this.fontProvider = builder.fontProvider
         this.listener = builder.listener
+        this.mode = builder.mode
         this.dismissRequest = builder.dismissRequest
     }
 
+    @Suppress("LongParameterList")
+    internal fun copy(
+        offeringSelection: OfferingSelection? = null,
+        shouldDisplayDismissButton: Boolean? = null,
+        fontProvider: FontProvider? = null,
+        listener: PaywallListener? = null,
+        mode: PaywallMode? = null,
+        dismissRequest: (() -> Unit)? = null,
+    ): PaywallOptions {
+        return Builder(dismissRequest ?: this.dismissRequest)
+            .apply {
+                when (val offeringSelectionToCheck = offeringSelection ?: this@PaywallOptions.offeringSelection) {
+                    is OfferingSelection.OfferingId -> setOfferingId(offeringSelectionToCheck.offeringId)
+                    is OfferingSelection.OfferingType -> setOffering(offeringSelectionToCheck.offeringType)
+                    is OfferingSelection.None -> { /* Do nothing */ }
+                }
+            }
+            .setShouldDisplayDismissButton(shouldDisplayDismissButton ?: this.shouldDisplayDismissButton)
+            .setFontProvider(fontProvider ?: this.fontProvider)
+            .setListener(listener ?: this.listener)
+            .setMode(mode ?: this.mode)
+            .build()
+    }
+
     class Builder(
         internal val dismissRequest: () -> Unit,
     ) {
         internal var offeringSelection: OfferingSelection = OfferingSelection.None
         internal var shouldDisplayDismissButton: Boolean = false
```

Comment:
The default here is `false`? Should we use `DEFAULT_DISPLAY_DISMISS_BUTTON` everywhere?

Reply:
Well, I thought that when the developer is in charge of presenting the paywall, they should handle the navigation back as well.. but maybe it's ok to default to true, and just have them set it to false... 

I can change it for it to be the same everywhere though

Reply:
No that makes sense 👍🏻